rfcs/042-editable-programmatic-descriptions.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: editable_programmatic_descriptions
+- Start Date: 2021-07-27
+- RFC PR: [amundsen-io/rfcs#42](https://github.com/amundsen-io/rfcs/pull/42)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000)
+
+# Editable Programmatic Descriptions
+
+## Summary
+
+Programmatic descriptions are a useful way for Amundsen users to store and display business specific metadata about their tables in Amundsen. This feature allows for some customization of Amundsen without customizing the underlying software. Currently, this feature focuses on the machine generated metadata use case. This type of data can be extracted from the source system of truth and ingested into Amundsen via databuilder.
+
+The same need exists for business specific human generated metadata. This type of data does not exist in a machine generated form that can be easily extracted from a source system. Instead, users need to add/edit this metadata directly in the Amundsen UI. 
+
+## Motivation
+
+This feature allows users to configure their Amundsen installation to support one or more editable programmatic descriptions. These editable programmatic descriptions allow users to enrich the metadata displayed on the table details page with additional business specific human generated information. This feature will allow them to add/edit these programmatic descriptions directly in the UI.
+
+## Guide-level Explanation (aka Product Details)
+
+An editable programmatic description is the same as a regular programmatic description, except that it is maintained directly in the UI by a human instead of by a scheduled databuilder job. For example, users could configure an editable programmatic description with a source of data_usage_policy and then update the data_usage_policy on individual table detail pages with a link to the applicable data usage policy on their company's content management site.
+
+Similar to table descriptions, if a table is disabled for editing, then so are the editable programmatic descriptions.
+
+Databuilder jobs may still be useful on a one-off (i.e. non-scheduled basis) to seed Amundsen with manually curated data or perform one-off data conversations.
```

Comment:
Are you stating that databuilder will be the _only_ way to create these editable prog descs (where as once they are created they can be edited in the UI)? 

Also, can you quickly highlight the change to databuilder as well, in particular the prog desc model?

Reply:
TL;DR - users can add programmatic descriptions to a table via the UI (if in `PROGRAMMATIC_DISPLAY` as editable) and via datdabuilder (if they decide on their own to do that). the current proposal requires no changes to databuilder.

Once a programmatic description is configured as editable in the configuration of the frontend service in `PROGRAMMATIC_DISPLAY` then it will always appear in the UI on every table detail page with placeholder text. So once that happens, users can edit the editable programmatic description to add a value if one does not exist.

AFAIK databuilder currently allows users to add any programmatic description they desire, even if it's source is not configured in `PROGRAMMATIC_DISPLAY`. Of course, it will only be displayed in the UI if it is in `PROGRAMMATIC_DISPLAY`. Based on this understanding, there should be no changes required to support a one-off load via databuilder of editable programmatic descriptions.